tests/test_operation.py

```diff
@@ -2826,6 +2827,57 @@ class CustomOperator(qml.operation.Operator):
     assert qml.equal(new_op, CustomOperator(2.3, wires=0))
 
 
+@pytest.mark.parametrize(
+    "coeffs, obs",
+    [
+        (
+            [1.5, 0.5, 1, 1],
+            [qml.Identity(1), qml.Z(1) @ qml.Z(2), qml.X(1) @ qml.Y(2), qml.Hadamard(1)],
+        ),
+        ([0.5], [qml.X(1)]),
+        ([1], [qml.X(0) @ qml.Y(1)]),
+        ([0.5, 0.5], [qml.Identity(1), qml.Identity(1)]),
+        (
+            [0.0625, 0.0625, -0.0625, 0.0625, -0.0625, 0.0625, -0.0625, -0.0625],
+            [
+                qml.Hadamard(0) @ qml.X(1) @ qml.X(2) @ qml.Y(3),
+                qml.X(0) @ qml.X(1) @ qml.Y(2) @ qml.X(3),
+                qml.X(0) @ qml.Y(1) @ qml.X(2) @ qml.X(3),
+                qml.X(0) @ qml.Y(1) @ qml.Y(2) @ qml.Y(3),
+                qml.Y(0) @ qml.X(1) @ qml.X(2) @ qml.X(3),
+                qml.Y(0) @ qml.X(1) @ qml.Hadamard(2) @ qml.Y(3),
+                qml.Y(0) @ qml.Y(1) @ qml.X(2) @ qml.Y(3),
+                qml.Y(0) @ qml.Y(1) @ qml.Y(2) @ qml.Hadamard(3),
+            ],
+        ),
+        (
+            [-0.5, 0.4, -0.3, 0.2],
+            [qml.Identity(0, 1), qml.X(1) @ qml.Y(2), qml.Identity(1), qml.Z(1) @ qml.Z(2)],
+        ),
+        ([1], [qml.Hadamard(1)]),
+    ],
+)
+def test_convert_to_hamiltonian(coeffs, obs):
+    """Test that arithmetic operators can be converted to Hamiltonian instances"""
+
+    opmath_instance = qml.dot(coeffs, obs)
+    converted_opmath = convert_to_legacy_H(opmath_instance)
+
+    if isinstance(qml.simplify(opmath_instance), (Prod, SProd, Sum)):
```

Comment:
why do you need this?

Reply:
The `qml.simplify` function covers the case in which a simplified opmath instance turns out to be an `Observable.` For example:

```
coeffs = [0.5, 0.5]
obs = [qml.Identity(1), qml.Identity(1)]
opmath_instance = qml.dot(coeffs, obs)
type(opmath_instance), type(qml.simplify(opmath_instance))
```
Resulting in:
```
(pennylane.ops.op_math.sum.Sum, pennylane.ops.identity.Identity)
```
If the instance is an `Observable` it is not converted by the function

Reply:
Thanks for elaborating, that sounds like in those cases nothing is tested though, right? I would just remove those cases then, or am I still missing something? 😅

Reply:
There's just one such case in the test. To be honest, I just inserted it so that `Codecov` does not complain that the last line of the function is not covered by tests : )

Reply:
Ah gotcha. Instead of doing that you could just test that line explicitly in a separate test, e.g.

```
def test_convert_to_hamiltonian_trivial(self):
    opmath_was_active = qml.operation.active_new_opmath()
    qml.operation.disable_new_opmath()
    hamiltonian_instance = qml.Hamiltonian([0.5, 0.5], [X(0), Y(1)])
    
    res = convert_to_hamiltonian(hamiltonian_instance)
    assert res == hamiltonian_instance

    if opmath_was_active:
        qml.operation.enable_new_opmath()
    else:
        qml.operation.disable_new_opmath()
```